client/workspaces.js

```diff
@@ -135,6 +135,66 @@ class Workspaces {
         document.getElementById('currently-open').innerText = name;
     }
 
+    refreshWorkspace(object, remove, edit = null, rotation = null, tint = null ){
```

Comment:
This method signature is awful -- it's collapsing together four (five?) different behaviours behind various flags, has no documentation, and the logic flow doesn't make sense. 

I'm not sure why this method even exists -- just put the logic for each section into the corresponding operation.